### ISSUE TITLE
Communicate to assistive tech whether sidebar is open or closed 

### ISSUE THREAD

**Issue by klemay** [hypothesis/client#2055]
To support WCAG 2.0 Criterion 4.1.2, we need to ensure users know whether the content related to the sidebar toggle is collapsed or expanded.

In some cases the control is communicated in the same way, whether it’s expanded or collapsed. But even when the state is reflected correctly, communicating it as selected, the state is not clear. It is difficult to know based on the button label text alone (toggle or resize sidebar) whether ‘selected’ means the content is expanded, resized, or collapsed.

**Comment by robertknight** [comment]
> It is difficult to know based on the button label text alone (toggle or resize sidebar) whether ‘selected’ means the content is expanded, resized, or collapsed.

Would changing the label from "Toggle or resize sidebar" to "Show annotation sidebar" be sufficient? I would think that would make a) the meaning of the "selected" state more obvious and b) make it clear that we are talking about the annotation sidebar and not some other "sidebar" that the page might happen to have for other purposes.

**Comment by klemay** [comment]
@robertknight I would think in addition we would need to communicate the state of the sidebar. So if the label is "Show annotation sidebar" we'd want to follow with "sidebar is closed" or "sidebar is open" - this would be especially helpful for low-vision users who do a combo of looking at their screen and relying on info from screen readers (e.g. "I can't see the sidebar. Is it because selecting the button didn't work, or because my impairment prevents me from seeing it? If the former, I need to try the button again. If the latter, I should have my screen reader read the contents to me."

**Comment by robertknight** [comment]
https://github.com/hypothesis/client/pull/2102 changes the labels for the sidebar and highlight toggle buttons to "Show annotation sidebar" and "Show highlights" so that the meaning of the pressed state is more obvious (before it was "Toggle or resize sidebar", where the meaning of "pressed" was less obvious). I think this partially addresses this.

**Comment by klemay** [comment]
Yes that does get us closer to solving this! Seems like all that would be left is communicating to the user the current state of the sidebar.

**Comment by robertknight** [comment]
> So if the label is "Show annotation sidebar" we'd want to follow with "sidebar is closed" or "sidebar is open" 

The screen reader will indicate that the button is a toggle button and say "selected" or "deselected" (or similar) after reading the label, so I would hope that would make it clear what the current state is, if paired with an appropriate label.

We could have a look at some popular applications that have similar sidebars. Google Drive has one on the right edge of the screen. It has a toggle button which is similar to ours except that the label changes between "Show side panel" and "Hide side panel" depending on the state.

**Comment by klemay** [comment]
I like the idea of the label changing between "Show annotation sidebar" and "Hide annotation sidebar" depending on the state. I think it's clearer than having "Show annotation sidebar" and following with "selected"/"deselected".
